### PR TITLE
DAT-61513 App Components | Tabs & Carousel Hooks refactoring 

### PR DESCRIPTION
**Pull request by mlevi1806**
## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Are you adding a new package?
- [ ] yes
- [x] no

## PR Checklist
- [ ] I'm using the theme colors from styled-components and checked my component in dark mode as well
- [ ] Add types (Prop-types / typescript)
- [ ] Commit your changes
- [ ] Run `npm version **` [minor, major, patch]. Commit changes
- [ ] Run `npm run lib:build`. Commit changes
- [ ] Run `npm run lib:version`. Commit changes
- [ ] Push changes and open a pull request

### CODE REVIEW

File: app/src/components/base/Tabs.js [hunk 1]
```diff
@@ -1,93 +1,90 @@
...
-    tabWidths: [],
-    prevWidth: []
-  };
+const Tabs = props => {
+  const [tabWidths, setTabsWidths] = useState([]);
+  const [prevWidth, setPrevWidth] = useState([]);
+  const [localWidths, setLocalWidths] = useState([]);
 
-  localWidths = [];
+  const calculateWidths = useCallback(() => {
```
Comment (crazypixel) [inline app/src/components/base/Tabs.js:10]: can you move this to the useEffect? and add to the dep. array - props.tabs.length

File: app/src/components/base/Tabs.js [hunk 2]
```diff
@@ -1,93 +1,90 @@
...
+  const {
+    contentRenderer,
+    labelRenderer,
+    selectedIndex,
+    tabs,
+    gap = 35,
+    onSelect,
+    justify,
+    className
+  } = props;
```
Comment (crazypixel) [inline app/src/components/base/Tabs.js:46]: space between props and return

File: app/src/components/base/Tabs.js [hunk 3]
```diff
@@ -1,93 +1,90 @@
...
+        <InnerHeader>
+          {tabs.map(tab => {
+            const key = `tab-${tab.id}`;
+            const handleClick = () => onSelect(tab.id);
+            const selected = tab.id === selectedIndex;
+
+            return (
+              <Tab
+                gap={gap / 2}
+                className={`tab-${tab.id}`}
```
Comment (crazypixel) [inline app/src/components/base/Tabs.js:59]: you can use the key var here

**Comment by Chikel** [comment]
@crazypixel , it looks like you're not this repo's CRA (change request approver), open a devops on-call task.

**Pull request merged**